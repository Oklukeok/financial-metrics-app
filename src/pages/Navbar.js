import React from 'react';
import logo from '../assets/img/logo.svg';

const Navbar = () => (
  <nav>
    <img className="logo" src={logo} alt="logo" />
    <h1>Crypto Realm</h1>
  </nav>
);

export default Navbar;
