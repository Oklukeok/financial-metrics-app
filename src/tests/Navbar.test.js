import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../pages/Navbar';
import coinsStore from '../Redux/configureStore';
import '@testing-library/jest-dom';

describe('Test for Navbar component', () => {
  test('Renders Navbar correctly', () => {
    const navbar = render(
      <Provider store={coinsStore}>
        <Router>
          <Navbar />
        </Router>
      </Provider>,
    );
    expect(navbar).toMatchSnapshot();
  });
});
