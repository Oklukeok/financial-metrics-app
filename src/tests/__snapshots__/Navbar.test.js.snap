// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Navbar page Should match the navbar snap 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <img
          alt="logo"
          class="logo"
          src="logo.svg"
        />
        <h1>
          Crypto Realm
        </h1>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav>
      <img
        alt="logo"
        class="logo"
        src="logo.svg"
      />
      <h1>
        Crypto Realm
      </h1>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
